tests/stdlib/tstrutils.nim

```diff
@@ -221,6 +221,86 @@ template main() =
     doAssert "0123456789ABCDEFGH".find({'A'..'C'}, 5, 10) == 10
     doAssert "0123456789ABCDEFGH".find({'A'..'C'}, 5, 9) == -1
 
+    block:
+      const haystack: string = "ABCABABABABCAB"
+      doAssert haystack.len == 14
+      doAssert haystack.find("ABC") == 0
+      doAssert haystack.find("ABC", last=13) == 0
+      doAssert haystack.find("ABC", last=12) == 0
```

Comment:
avoid redundant tests, it gives the illusion of increased test coverage where in reality the same code path is being tested.

what's the point of `doAssert haystack.find("ABC", last=12) == 0` for e.g.?
likewise in lots of other places.

ideally, tests cover each code path, and boundary conditions, but should still not be redundant.

Reply:
I removed few test cases that I agree were redundant, but some of these test cases do test valid boundaries, like `doAssert haystack.find("ABC", last=12) == 0`. In this case, I see passing `len("ABC")`, `high("ABC")`, and `high("ABC")-1` into `last` as valid tests because they are either at a boundary or off by one from a boundary.

The is further complicated by the fact that the default value for `last` in the `find` methods is 0, which is a valid index. IMO the default should be -1 (as it is for `rfind`). This now makes each of -1, 0, and +1 either a boundary or off by one from a boundary.

I feel that blocks like this, for example https://github.com/nim-lang/Nim/blob/0b5346de87fda6136fc91a02ae9700078fc95679/tests/stdlib/tstrutils.nim#L244-L246 are warranted because index 9 is the start of `"ABC"` and index 11 is the end. Therefore, 11 is a boundary, and 12 and 10 are off by one from that boundary.